### PR TITLE
locale: avoid TOCTOU in reading config files

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/shared/locale-setup.c [hunk 1]
```diff
@@ -208,11 +212,9 @@ int locale_context_save(LocaleContext *c, char ***ret_set, char ***ret_unset) {
         if (r < 0)
                 return r;
 
-        if (stat("/etc/locale.conf", &st) < 0)
+        if (stat("/etc/locale.conf", &c->st) < 0)
```
Comment (keszybz) [inline src/shared/locale-setup.c:215]: Hmm, but this is also a TOCTOU race. Maybe it's not worth fixing…

I think we could have `write_env_file_full()` that'd take a `struct stat *` and return that stat of the fd after `flush_and_check()`.

**Review by keszybz — CHANGES_REQUESTED** [review]
LGTM. That one comment, feel free to ignore it and upgrade to `waiting-for-ci`.

File: src/shared/locale-setup.c (hunk 1 above)
Comment (yuwata) [reply src/shared/locale-setup.c:215]: Yeah, possibly. I will try to address that later in another PR.

**Pull request merged**